connectivity/mqtt/client_common.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2019 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#ifndef CLIENT_CONFIG_H
+#define CLIENT_CONFIG_H
+
+#include <stdio.h>
+#include "accelerator/errors.h"
+#include "third_party/mosquitto/lib/mosquitto.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file connectivity/mqtt/client_common.h
+ */
+
+#define ID_LEN 32
+#define API_NUM 9
+
+typedef enum client_type_s { client_pub, client_sub, client_duplex } client_type_t;
+
+typedef enum mosq_err_t mosq_retcode_t;  // typedef the original enum
+
+typedef struct mosq_general_config_s {
+  char *id;
+  int protocol_version;
+  int keepalive;
+  char *host;
+  int port;
+  int qos;
+  int last_mid;
+  bool retain;
+  client_type_t client_type;
+#ifdef WITH_SRV
```

Comment:
Can you explain how it works with `WITH_SRV`?

Reply:
I haven't investigated this function provided bt mosquitto yet. I just keep them at this moment, in case I will use them in the future.
Other functionalities such as TLS and SOCKS haven't been supported at this moment as well.
But I think it will be pretty useful if we support them in the future.

Reply:
Never add the code you have not verified and fully understood.